### PR TITLE
Refactor borrow book classes

### PR DESCRIPTION
**Pull request by CameronNyb**

### CODE REVIEW

File: src/library/borrowbook/BorrowBookControl.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            
+        this.ui = ui;
+        ui.setState(BorrowBookUI.UI_STATE.READY);
+        state = CONTROL_STATE.READY;        
+    }
+        
+    public void swiped(int memberId) {
+        if (!state.equals(CONTROL_STATE.READY)) 
+            throw new RuntimeException("BorrowBookControl: cannot call cardSwiped except in READY state");
+
```
Comment (samuli64) [inline src/library/borrowbook/BorrowBookControl.java]: D2 brackets

File: src/library/borrowbook/BorrowBookControl.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    
+    public BorrowBookControl() {
+        this.library = Library.getInstance();
+        state = CONTROL_STATE.INITIALISED;
+    }
+
+    public void setUi(BorrowBookUI ui) {
+        if (!state.equals(CONTROL_STATE.INITIALISED)) 
+            throw new RuntimeException("BorrowBookControl: cannot call setUI except in INITIALISED state");
+
```
Comment (samuli64) [inline src/library/borrowbook/BorrowBookControl.java]: D1 Brackets

File: src/library/borrowbook/BorrowBookUI.java [hunk 3]
```diff
@@ -3,116 +3,99 @@
...
+        output("Borrow Book Use Case UI\n");
+
+        while (true) {
+            switch (state) {
+                case CANCELLED:
+                    output("Borrowing Cancelled");
+                    return;
+
+                case READY:
+                    String memStr = input("Swipe member card (press <enter> to cancel): ");
```
Comment (samuli64) [inline src/library/borrowbook/BorrowBookUI.java]: D3 memStr -> memberIdString

File: src/library/borrowbook/BorrowBookUI.java [hunk 4]
```diff
@@ -3,116 +3,99 @@
...
+                    break;
+
+                case SCANNING:
+                    String bookStringInput = input("Scan Book (<enter> completes): ");
+                    if (bookStringInput.length() == 0) {
+                        control.complete();
+                        break;
+                    }
+                    try {
+                        int bId = Integer.valueOf(bookStringInput).intValue();
```
Comment (samuli64) [inline src/library/borrowbook/BorrowBookUI.java]: D4 bId -> bookId

File: src/library/borrowbook/BorrowBookUI.java [hunk 5]
```diff
@@ -3,116 +3,99 @@
...
+                        control.scanned(bId);
+                        
+                    } catch (NumberFormatException e) {
+                        output("Invalid Book Id");
+                    } 
+                    break;
+
+                case FINALISING:
+                    String ans = input("Commit loans? (Y/N): ");
```
Comment (samuli64) [inline src/library/borrowbook/BorrowBookUI.java]: D5 ans -> answer

**Review by samuli64 — CHANGES_REQUESTED** [review]
Code reviewed, some defects identified. Please address before merge approval.

File: src/library/borrowbook/BorrowBookControl.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+import library.entities.Loan;
+import library.entities.Member;
+
+public class BorrowBookControl {
+
+    private BorrowBookUI ui;
+    
+    private Library library;
+    private Member member;
+    private enum CONTROL_STATE { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
```
Comment (samuli64) [inline src/library/borrowbook/BorrowBookControl.java]: D6 CONTROL_STATE -> ControlState

File: src/library/borrowbook/BorrowBookUI.java [hunk 7]
```diff
@@ -3,116 +3,99 @@
...
-
-
-	public void DiSpLaY(Object object) {
-		OuTpUt(object);		
-	}
-
-
-}
+
+    public static enum UI_STATE { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
```
Comment (samuli64) [inline src/library/borrowbook/BorrowBookUI.java]: D7 UI_STATE > UIState

**Comment by CameronNyb** [comment]
Defects 1-7 addressed

**Review by samuli64 — APPROVED** [review]
Reviewed. Defects remediated. Approval to merge.

**Pull request merged**